src/app/ngrx/effects/routing.effects.ts

```diff
@@ -25,14 +25,26 @@ export class RoutingEffects {
   // The router reducer captures the CustomRouterNavigationAction to save routing state.
   // The CustomRouterNavigationAction is mocked by tests but is _not_ otherwise ever manually called by the application,
   // only through routing and the router-store connecting module.
-  @Effect()
-  customRouterNavigation$: Observable<Action> = this.actions$
+  @Effect({dispatch: false})
+  customRouterNavigation$: Observable<void> = this.actions$
     .ofType(ROUTER_NAVIGATION)
     .map((action: ACTIONS.CustomRouterNavigationAction) => action.payload)
     .switchMap((payload: ACTIONS.CustomRouterNavigationPayload) => {
       const routerState: RouterModel = {...payload.routerState};
       // map to an action with our CUSTOM_ROUTER_NAVIGATION type
-      return Observable.of(new ACTIONS.CustomRouterNavigationAction({routerState}));
+      if (Object.keys(routerState).length > 0) {
+        // TODO: fix below and return Action rather than only sometimes dispatch
+        this.store.dispatch(new ACTIONS.CustomRouterNavigationAction({routerState}));
+      }
+      // TODO: #141 and #142
```

Comment:
This feels less good, but hopefully with all the TODOs I'm making on #141 and #142 I'll be able to re-address selectors soon and undo some weirdness